core/src/server/rpc_module.rs

```diff
@@ -56,22 +57,15 @@ use super::IntoResponse;
 /// implemented as a function pointer to a `Fn` function taking four arguments:
 /// the `id`, `params`, a channel the function uses to communicate the result (or error)
 /// back to `jsonrpsee`, and the connection ID (useful for the websocket transport).
-pub type SyncMethod = Arc<dyn Send + Sync + Fn(Id, Params, MaxResponseSize) -> MethodResponse>;
```

Comment:
another option is to add new `MethodKind` here such as:

```rust
RawMethod = Arc<dyn Send + Sync + Fn(Id, Params, ConnectionContext) -> MethodResponse>
```

then let folks spawn futures and stuff them selves if they need and just provide a new attribute in the procs macros as you already did.

then register_blocking, async, etc are not needed.

Reply:
then we don't have break existing APIs

Reply:
That makes sense, I've implemented that in: https://github.com/paritytech/jsonrpsee/pull/1297/files#diff-dc533588250ea236f580d7fdaf87cdfce2ec80da031de9067b1088465a555998R64

One downside with doing a `RawMethod` like this is that we lose the ability to have:
- async methods (since this is a `SyncMethod` that receives an extra connection ID)
- blocking methods  (since the server-context is passed differently to a blocking method -- its need to be `Arc<Context>` instead of `&Context` so our callbacks are different)

This makes me wonder if I should also add the counter-part RawAsyncMethod and another `register_raw_blocking`. I think a single `RawMethod` would solve the connection ID exposure to complete the chainHead issue in substrate; but it might require us to extend it in the future.

At some point in the future , if we choose to implement `RawAsyncMethod`, it might be more beneficial to make a breaking change and add the `connection Id` to every method, for now I think we could avoid that